### ISSUE TITLE
Support extensions for request object

### ISSUE THREAD

**Issue by 0x6373** [ybbus/jsonrpc#30]
Hi, not sure if this is the right place to ask but i'm not sure what else to do.

Is it planned to support custom additional members in the request object?
Specifically i would like to use this jsonrcp client to communicate with Zabbix (Monitoring Software) since it states that it implement an JSONRPC API (https://www.zabbix.com/documentation/4.2/manual/api).

From my point of view this is not true since they require an additional 'auth' member in the request object containing a token which is aquired with the 'user.login' method (https://www.zabbix.com/documentation/4.2/manual/api)

I also found another tool that requires something like that (http://documentation.solarwindsmsp.com/backup/documentation/Content/service-management/json-api/login.htm)

I got no answer from Zabbix via IRC and forum and found some mentions of jsonrpc extensions, but found no further documentation on how these are implemented.

Long story short, is it planned to to support custom member in the request object like there is support for custom headers or a custom http client?

**Comment by ybbus** [comment]
Hi.

In my opinion this is no valid json rpc regarding the specification.
You can argue if you may provide additional fields, but if some components are very strict about json rpc format, this may cause problems.

Another strong opinion is, that authentication information belongs into the header, not the body (see. basic auth, oauth tokens etc.).

I think the easiest way would be for you to fork the project and add the extra field you need.

Short answer: there is no custom parameters planned until there is a lot of demand, sorry.

**Comment by 0x6373** [comment]
Thank you for the response, i agree with your point-of-view, but i was unsure about those jsonrcp extensions.
